sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala

```diff
@@ -268,7 +268,7 @@ private object CosmosAccountConfig {
       } else {
         trimmedInput.split(",")
           .toStream
-          .map(preferredRegion => preferredRegion.toLowerCase(Locale.ROOT).replace(" ", ""))
+          .map(preferredRegion => preferredRegion.toLowerCase(Locale.ROOT).trim)
```

Comment:
The java SDK is not trimming. It just fails silently. 
Java SDK expects the region in the format "North Central US" (case insensitive). If specified in any other format it will silently default to the correctly specified next available read region. If no read region is specified correctly then it will use the primary region.

Reply:
As discussed offline , we will go with consistency with sdk (java/.net) , and trimming  or any other altering on preferred name should have separate discussion and then reflect back on all the connector at once.

Reply:
I don't understand this discussion. There is no need to keep config in-sync between Spark config (always text based) and Java SDK. So I think we should use the config parsing as it was intended to be (and documented in config reference) - which I believe is what Ravi's PR is doing here. Let's not spend any more time on this - as long as we have regression coverage :-) - which we do have now I believe

Reply:
Our documentation always has examples with leading or trailing space.                                    
https://docs.microsoft.com/en-us/azure/cosmos-db/sql/tutorial-global-distribution-sql-api?tabs=dotnetv2%2Capi-async#java4-preferred-locations
I think we should go with trimming in the spark and update the java SDK in the next release.

Reply:
Thanks @RaviTella  make sense